utils/create_dataset.py

```diff
@@ -104,3 +114,155 @@ def __getitem__(self, index):
                                 f"\nIgnore if some augmentations have padded with dontcare value.")
         sample['index'] = index
         return sample
+
+
+class VRTDataset(GeoDataset):
+    def __init__(self, vrt_ds: DatasetReader) -> None:
+        """Initialize a new VRTDataset instance.
+        The dataset is base on the DataReader class, initiated by rasterio.open().
+
+        Args:
+            vrt_ds: DatasetReader object (rasterio)
+        """
+        super().__init__()
+
+        self.vrt_ds = vrt_ds
```

Comment:
You could use check_rasterio_im_load here and therefore accept a "vrt_ds" (maybe rename this argument if applying this change) that could be either a file directly (as torchgeo implements out-of-the-box) or DatasetReader instance like we want.

This would make your class look close to something we could suggest to the torchgeo folks in a further issue.